src/main/java/seedu/address/ui/TagStatisticsPanel.java

```diff
@@ -0,0 +1,98 @@
+package seedu.address.ui;
+
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+import java.util.logging.Logger;
+
+import com.google.common.eventbus.Subscribe;
+
+import javafx.application.Platform;
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import javafx.fxml.FXML;
+import javafx.scene.chart.BarChart;
+import javafx.scene.chart.CategoryAxis;
+import javafx.scene.chart.XYChart;
+import javafx.scene.layout.Region;
+import seedu.address.commons.core.LogsCenter;
+import seedu.address.commons.events.model.AddressBookChangedEvent;
+import seedu.address.model.ReadOnlyAddressBook;
+import seedu.address.model.person.ReadOnlyPerson;
+import seedu.address.model.tag.Tag;
+
+//@@author Valerieyue
+/**
+ * The birthday statistics panel of the App.
+ */
+public class TagStatisticsPanel extends UiPart<Region> {
+
+    private static final String FXML = "TagStatisticsPanel.fxml";
+
+    @FXML
+    private BarChart<String, Integer> barChart;
+
+    @FXML
+    private CategoryAxis xAxis;
+
+    private final Logger logger = LogsCenter.getLogger(this.getClass());
+    private ObservableList<String> tagNames = FXCollections.observableArrayList();
+
+    public TagStatisticsPanel(ReadOnlyAddressBook readOnlyAddressBook) {
+        super(FXML);
+        setPersonData(readOnlyAddressBook);
+        barChart.setCategoryGap(10);
+        barChart.setTitle("Tags Statistics");
+        xAxis.setLabel("Tags");
+        registerAsAnEventHandler(this);
+    }
+
+    private void setPersonData(ReadOnlyAddressBook readOnlyAddressBook) {
+        XYChart.Series<String, Integer> series = new XYChart.Series<>();
+
+        // Get an array with the tags.
+        tagNames.clear();
+        for (ReadOnlyPerson person : readOnlyAddressBook.getPersonList()) {
+            Set<Tag> personTags = person.getTags();
+            for (Tag tag : personTags) {
+                // to prevent adding duplicated tags
+                if (!tagNames.contains(tag.getTagName()))
+                    tagNames.add(tag.getTagName());
+            }
+        }
+        Logger.getLogger(tagNames.toString());
+
+        // Assign the tag as categories for the horizontal axis.
+        xAxis.setCategories(tagNames);
+
+        // Count the number of people who have the same tag.
+        int[] tagCounter = new int[tagNames.size()];
```

Comment:
here you can try to use HashMap<key, value>, where key could be the tag and value is the counter. so in the for loop, you just have to use check if it contains the key, if it does then ++.